### DISCUSSION TITLE
Tycho and Junit 5

### DISCUSSION THREAD

**Discussion by cdietrich** [eclipse-tycho/tycho#1259] [Q&A] [answered]
as consequence of  https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/449 i try to switch from requirebundle to import package for org.junit.jupiter.api
but the test dont seem to be executed as junit5 tests anymore.

any hints where to start digging?

**Comment by akurtakov** [comment]
Which Tycho version do you use? Tycho 3.0.0-SNAPSHOT has the knowledge about new bundle names.

**Reply by cdietrich** [reply]
2.7.4. we use custom extensions. will try to add an explict import to org.junit.jupiter.api.extension;version="[5.1.0,6.0.0) as well
for unknown reasons we also have junit5 container on classpath
so that it compiled without it.

**Reply by laeubi** [reply]
> for unknown reasons we also have junit5 container on classpath
> so that it compiled without it.

This is actually desired and tycho simply do not detect this reliable enough yet...

**Reply by cdietrich** [reply]
adding the import to extension seems to work
i am not sure if and when https://github.com/eclipse/xtext-core/issues/666#issuecomment-389567981 is still needed though

**Comment by cdietrich** [comment, marked as answer]
The problem was solved by not only importing api but also api.extension.
There was no compile error cause we had additionally the Junit5 container configured and tycho seems to pick it up at compile time
